app/build.gradle

```diff
@@ -40,6 +48,31 @@ dependencies {
     implementation 'com.google.android.material:material:1.3.0'
     implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
     testImplementation 'junit:junit:4.+'
+    implementation 'androidx.core:core-ktx:1.3.1'
+
+    // Room and Lifecycle dependencies
+    implementation "androidx.room:room-runtime:2.3.0"
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```

Comment:
Why this dependency is needed ?

Reply:
What part of code needed this dependency androidx.legacy:legacy-support-v4:1.0.0